server/src/main/java/org/candlepin/resource/HypervisorResource.java

```diff
@@ -123,22 +125,42 @@ public HypervisorCheckInResult hypervisorUpdate(
 
         Owner owner = this.getOwner(ownerKey);
 
+        if (hostGuestMap.remove("") != null) {
+            log.warn("Ignoring empty hypervisor id");
+        }
+
         // Maps virt hypervisor ID to registered consumer for that hypervisor, should one exist:
         VirtConsumerMap hypervisorConsumersMap =
                 consumerCurator.getHostConsumersMap(owner, hostGuestMap.keySet());
 
+        int emptyGuestIdCount = 0;
+        int emptyHypervisorIdCount = 0;
         Set<String> allGuestIds = new HashSet<String>();
-        for (Entry<String, List<GuestId>> hostEntry : hostGuestMap.entrySet()) {
-            for (GuestId gid : hostEntry.getValue()) {
-                allGuestIds.add(gid.getGuestId());
+        Iterator<Entry<String, ArrayList<GuestId>>> hostEntries = hostGuestMap.entrySet().iterator();
+        while (hostEntries.hasNext()) {
+            Entry<String, ArrayList<GuestId>> hostEntry = hostEntries.next();
+            ListIterator<GuestId> guestIds = hostEntry.getValue().listIterator();
+            while (guestIds.hasNext()) {
+                GuestId gid = guestIds.next();
+                String guestId = gid.getGuestId();
+                if (!guestId.equals("")) {
+                    allGuestIds.add(guestId);
+                }
+                else {
+                    emptyGuestIdCount++;
+                    guestIds.remove();
+                }
             }
         }
+        if (emptyGuestIdCount > 0) {
+            log.warn("Found " + emptyGuestIdCount + " empty guest id(s). Ignoring.");
```

Comment:
Avoid string concats in log statements.

Use

```java
log.warn("Found {} empty guest id(s). Ignoring.", emptyGuestIdCount);
```

See [here](http://www.slf4j.org/manual.html)